src/setup/extract-configuration.go

```diff
@@ -94,17 +101,29 @@ func ExtractConfiguration(configFilePath string) Configuration {
 	if parsedConfig.Provider == "" {
 		parsedConfig.Provider = defaultProvider
 	}
+	if parsedConfig.Runtime == "" {
+		parsedConfig.Runtime = defaultRuntime
+	}
 
 	for index := range parsedConfig.SubExperiments {
 		if parsedConfig.SubExperiments[index].Function == "" {
 			parsedConfig.SubExperiments[index].Function = defaultFunction
 		}
+		if parsedConfig.SubExperiments[index].Handler == "" {
+			parsedConfig.SubExperiments[index].Handler = defaultHandler
+		}
+		if parsedConfig.SubExperiments[index].Runtime == "" {
```

Comment:
Could you delete the default assignment for the subexperiment runtime? Take an example: You want to have 5 python subexperiments (and no other runtime). Now, you have to specify in the JSON that you want python in all subexperiments, instead of specifying it only on the provider level. 
The deafult for this field should be either nothing (as it is an optional field which overwrites the provider runtime) or it should be the same as the provider level runtime (parsedConfig.Runtime).

Reply:
Would you prefer we do:
```go
if parsedConfig.SubExperiments[index].Runtime == "" {
	parsedConfig.SubExperiments[index].Runtime = parsedConfig.Runtime
}
```

Or just remove the if statement in the subexperiment?

I am thinking that the former might be better, as when creating the serverless.yml file, we can do:

```diff go
func (s *Serverless) AddFunctionConfig(subex *SubExperiment, index int) {
	// ...

	for i := 0; i < subex.Parallelism; i++ {
		// ...
-               f := &Function{Handler: subex.Handler, Name: name, Events: events}
+               f := &Function{Handler: subex.Handler, Name: name, Events: events, Runtime: subex.Runtime}
		s.Functions[name] = f
		subex.addRoute(name)
	}

}
```